platform/pulp/plugin/changeset/report.py

```diff
@@ -0,0 +1,69 @@
+class Action:
+    ADDED = 'ADDED'
+    DELETED = 'DELETED'
+
+
+class Disposition:
+    PENDING = 'PENDING'
+    SUCCEEDED = 'SUCCEEDED'
+    FAILED = 'FAILED'
+
+
+class ChangeReport:
+    """
+    Report changes to a repository.
+
+    Attributes:
+        action (str): The action taken (ADD, DEL).
+        content (pulp.plugin.Content): The affected content model.
+        errors (list): A list of reported errors.
+    """
+
+    def __init__(self, action, content):
+        """
+        Args:
+            action (str): The action taken (ADD, DEL).
+            content (pulp.plugin.Content): The affected content model.
+        """
+        self.action = action
+        self.content = content
+        self.errors = []
+
+    @property
+    def disposition(self):
+        """
+        The reported disposition.
+
+        Returns:
+            (str):
+        """
+        if self.errors:
+            return Disposition.FAILED
+        else:
+            return Disposition.SUCCEEDED
+
+
+class AddReport(ChangeReport):
+    """
+    Report of added content.
+    """
+
+    def __init__(self, content):
+        """
+        Args:
+            content (pulp.plugin.Content): The added content model.
+        """
+        super(AddReport, self).__init__(Action.ADDED, content)
```

Comment:
I wonder if you need a whole class just for this. You end up being able to identify that a ChangeReport either added or deleted in two different ways. You can check its ``action`` attribute, and you can check if it is an instance of ``AddReport`` or ``DeleteReport``. I don't think you need both of these, so I would lean toward ditching the extra classes and just use the attribute.